### PR TITLE
Allow custom endpoints

### PR DESCRIPTION
**Pull request by Briankaiser**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -126,10 +140,19 @@ export function createActionLogger(options: ActionLoggerOptionsType): Function {
   if (endpoint === null || endpoint === undefined) {
     throw new Error('endpoint is required');
   }
-  if (!endpoint.uri || (typeof endpoint.uri !== 'string') || endpoint.uri.length === 0) {
-    throw new Error('endpoint.uri must be a valid string');
-  }
 
+  let endpointFunction;
+  if (typeof endpoint === 'function') {
+    if (endpoint.length != 2) {
```
Comment (agilejon) [inline src/index.js:152]: Nice, I didn't know `.length` was a property of functions. That's cool

File: src/index.js [hunk 2]
```diff
@@ -33,56 +35,68 @@ export type ActionLoggerOptionsType = {
...
+    let finalHeaders = {};
+    let finalLog = log;
+
+    if (headers) {
+      const mergedHeaders = {...headers};
+
+      // go through each header. if function - call and replace value
+      for (const key in endpoint.headers) {
+        if (typeof endpoint.headers[key] === 'function') {
+          mergedHeaders[key] = (endpoint.headers[key].call(undefined, state): string);
```
Comment (agilejon) [inline src/index.js]: Could do this in a future PR, but a try/catch here feel warranted. Maybe just set the header to something like `"Exception generating header"` if it fails

File: src/index.js [hunk 3]
```diff
@@ -174,19 +197,14 @@ export function createActionLogger(options: ActionLoggerOptionsType): Function {
       }
     }
 
-
-    if (endpoint.transformFunction) {
```
Comment (agilejon) [inline src/index.js:178]: Moving this makes sense, the transform may not be necessary if there's a custom endpoint function

**Review by agilejon — APPROVED** [review]
One minor suggestion and two 'oh cool' comments. This looks good.

File: src/index.js (hunk 2 above)
Comment (Briankaiser) [reply src/index.js]: makes sense. I just pushed an update (and associated test) - does that meet your expectations?

**Pull request merged**